app/controllers/users_controller.rb

```diff
@@ -21,10 +29,48 @@ def create
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
+    if @user.update user_params
+      flash[:success] = t "edit_user.user_updated"
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    User.find_by(id: params[:id]).destroy
+    flash[:success] = t "users.user_deleted"
+    redirect_to users_path
+  end
+
   private
   def user_params
     # Strong Parameters
     params.require(:user)
           .permit :name, :email, :password, :password_confirmation
   end
+
+  def logged_in_user
+    return if logged_in?
+
+    store_location
+    flash[:danger] = t "please_login"
+    redirect_to login_path
+  end
+
+  def correct_user
+    @user = User.find_by id: params[:id]
+    redirect_to root_path unless current_user? @user
+  end
+
+  def admin_user
+    flash[:danger] = t "action_not_allowed"
+    redirect_to root_path unless current_user.admin?
```

Comment:
code ri có ổn ko, dù có là admin ko thì cũng có flash danger à =))

```suggestion
    return if current_user.admin?
    
    flash[:danger] = t "action_not_allowed"
    redirect_to root_path
```